### PR TITLE
[language] Add FieldHandle to proptest

### PR DESCRIPTION
**Pull request by dariorussi**
This PR adds the ability for proptest valid strategy to generate
non generic field instructions.
That extend test coverage for binary format tests and verifier
consistencies checks.
It also starts plugging in generic instructions.

### CODE REVIEW

**Comment by dariorussi** [comment]
this is a work in progress but it's also extending test coverage so I am happy about how it is growing.
It needs a lot more cleanup but a lot of that will be dictated by how we add generic which is coming next

File: language/vm/src/proptest_types.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@ mod signature;
 use crate::proptest_types::functions::{FnHandleMaterializeState, FunctionHandleGen};
 use functions::{FnDefnMaterializeState, FunctionDefinitionGen};
 use signature::{KindGen, SignatureTokenGen};
-use std::collections::{BTreeSet, HashSet};
+use std::collections::{BTreeSet, HashMap, HashSet};
```
Comment (sblackshear) [inline language/vm/src/proptest_types.rs:27]: `TreeMap` to avoid accidental nondeterminism from iterating over this?

Comment (dariorussi) [reply language/vm/src/proptest_types.rs:27]: I may change it at some point just for "safety".
The point is that right now those are literally used for hash and not iteration at all. Basically they are used to keep track of some binary value to its index...

**Pull request closed without merging**